lstchain/reco/volume_reducer.py

```diff
@@ -0,0 +1,146 @@
+"""
+Module with functions to perform a volume reduction to the dl1 files:
+NOTE : The volume reduction is indeed applied in the process
+between R1 to dl1.
+
+Usage:
+
+"import volume_reducer"
+"""
+import numpy as np
+from copy import deepcopy
+from ctapipe.image.cleaning import tailcuts_clean, dilate
+
+__all__ = ['apply_volume_reduction',
+           'zero_suppression_tailcut_dilation',
+           'dilate_pixel_mask',
+           'none_volume_reduction'
+           ]
+
+
+def apply_volume_reduction(event, tel_id, config):
+    """
+    Function that looks into the loaded configuration and applies to the data (modifying
+    the event container) the volume reduction method to be applied.
+    A `none_volume_reduction` function is applied by default.
+
+    :param event: [source event. ctapipe datacontainer]
+            An event of a loaded source. A 'ctapipe.io.containers.DataContainer' object.
+    :param tel_id: [int]
+            A telescope id wih data: an `event.r0.tels_with_data` object.
+    :param config: [dict]
+            Loaded configuration: from where the algorithm (volume reduction method)
+            and their corresponding parameters are loaded.
+
+    :return: none.
+            Modifies the event containers by applying the computed mask to the image,
+            the waveform and the pulse_time objects, as:
+            image[~mask] = 0, ...
+    """
+    camera = event.inst.subarray.tel[tel_id].camera
+    image = event.dl1.tel[tel_id].image
+    pulse_time = event.dl1.tel[tel_id].pulse_time
+    waveform = event.r1.tel[tel_id].waveform
+
+    if config['volume_reducer']['algorithm'] is None:
```

Comment:
I know I am the one who suggested to have the same function structure even for no volume reduction but that might have been a bad suggestion. It would be more efficient to just
```
if config['volume_reducer']['algorithm'] is None:
     pass
```